### PR TITLE
Consolidate node provisioning tests

### PR DESCRIPTION
**Pull request by basil**
`QueueTest` contains three node-provisioning tests: `flyweightTasksWithoutMasterExecutors`, `shouldRunFlyweightTaskOnProvisionedNodeWhenNodeRestricted`, and `shouldBeAbleToBlockFlyweightTaskAtTheLastMinute`. The last of these has been `@Ignored` for a very long time, while the second of these contains to flake as below:

```
02:21:15  [ERROR] hudson.model.QueueTest.shouldRunFlyweightTaskOnProvisionedNodeWhenNodeRestricted  Time elapsed: 180.066 s  <<< ERROR!
02:21:15  org.junit.runners.model.TestTimedOutException: test timed out after 180 seconds
02:21:15        at java.base@11.0.16/java.lang.Object.wait(Native Method)
02:21:15        at java.base@11.0.16/java.lang.Object.wait(Object.java:328)
02:21:15        at app//hudson.remoting.AsyncFutureImpl.get(AsyncFutureImpl.java:78)
02:21:15        at app//org.jvnet.hudson.test.JenkinsRule.assertBuildStatus(JenkinsRule.java:1479)
02:21:15        at app//org.jvnet.hudson.test.JenkinsRule.buildAndAssertStatus(JenkinsRule.java:1526)
02:21:15        at app//org.jvnet.hudson.test.JenkinsRule.buildAndAssertSuccess(JenkinsRule.java:1512)
02:21:15        at app//hudson.model.QueueTest.shouldRunFlyweightTaskOnProvisionedNodeWhenNodeRestricted(QueueTest.java:1027)
02:21:15        at java.base@11.0.16/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
02:21:15        at java.base@11.0.16/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
02:21:15        at java.base@11.0.16/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
02:21:15        at java.base@11.0.16/java.lang.reflect.Method.invoke(Method.java:566)
02:21:15        at app//org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
02:21:15        at app//org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
02:21:15        at app//org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
02:21:15        at app//org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
02:21:15        at app//org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
02:21:15        at app//org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:613)
02:21:15        at app//org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
02:21:15        at app//org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
02:21:15        at java.base@11.0.16/java.util.concurrent.FutureTask.run(FutureTask.java:264)
02:21:15        at java.base@11.0.16/java.lang.Thread.run(Thread.java:829)
```

Since I do not trust `NodeProvisionerRule` to be a realistic test scenario, before I start debugging these I am moving `flyweightTasksWithoutMasterExecutors` and `shouldRunFlyweightTaskOnProvisionedNodeWhenNodeRestricted` to `NodeProvisionerTest` where they can use `RealJenkinsRule`, thereby eliminating the last usage of `NodeProvisionerRule` in Jenkins core. I have not bothered with `shouldBeAbleToBlockFlyweightTaskAtTheLastMinute` as it has not been a functional test for a long time; if someone cares to resurrect it, they can do so from the Git history. I am skipping the `flyweightTasksWithoutMasterExecutors` and `shouldRunFlyweightTaskOnProvisionedNodeWhenNodeRestricted` on Windows for consistency with the other tests in `NodeProvisionerTest`.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  - Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7210"></a>

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
